File: app/src/main/java/com/gpetuhov/samplemockito/MainActivity.java
package com.gpetuhov.samplemockito;

import android.os.Bundle;
import android.widget.TextView;

import com.gpetuhov.samplemockito.utils.PrefsUtils;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private PrefsUtils prefsUtils;
    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        prefsUtils = new PrefsUtils();

        prefsUtils.setText( "Some text bla bla bla" );

        textView = findViewById( R.id.textview );

        textView.setText( prefsUtils.getText() );

        // TODO: Test PrefUtils
        // TODO: Shared prefs must be mocked with Mockito
    }
}

File: app/src/main/java/com/gpetuhov/samplemockito/MainActivity.kt
package com.gpetuhov.samplemockito

import android.os.Bundle
import android.widget.TextView

import com.gpetuhov.samplemockito.utils.PrefsUtils

import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    @Inject
    lateinit var prefsUtils: PrefsUtils

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        SampleMockitoApp.dataComponent.inject(this)

        prefsUtils.text = "Some text bla bla bla"

        textView.text = prefsUtils.text

        // TODO: Test PrefUtils
        // TODO: Shared prefs must be mocked with Mockito
    }
}
